[Security] package bumps -- possible breaking change

yuenmichelle1 opened the pull request:
this started off with Django update to 3.2.13, but while updating django, i realzed there were too many packages conflicting. 

easiest was to start fresh and work backwards to see what package was breaking the updates. that led to updating all packages theia uses.
unfortunately, until USGS api allows us entry to their api again, i cannot test repercussions (if any) to the whole floating forests pipeline, SO I would **WAIT** until after we get access again to merge in this change.

there are also a lot of warnings due to package updates (a lot of methods that we use will/might be deprecated at some point upon next version bump, so we may want to take care of that). i'll try to take care of some in this PR.

chelseatroy commented:
I realize this is a WIP, and I agree with your intuition to wait and see that this works against the USGS API before merge. But I went ahead and took a sneak peek :) 

I'm a fan of the rename in settings. I'm curious about this line:

`django_find_project = false `

I actually realize I don't know what this does. What is this thingy? Why did we need to change it?

yuenmichelle1 commented:
> I realize this is a WIP, and I agree with your intuition to wait and see that this works against the USGS API before merge. But I went ahead and took a sneak peek :)
> 
> I'm a fan of the rename in settings. I'm curious about this line:
> 
> `django_find_project = false `
> 
> I actually realize I don't know what this does. I looked it up to get a sense of what it does _in general_, but I'd be curious what forced us to change it in our use case, especially because that knowledge might be helpful for dependency troubleshooting down the road.

https://pytest-django.readthedocs.io/en/latest/managing_python_path.html

This was an interesting one and its probably because I updated `pytest` when I updated the 💩 ton of packages on theia. I think the update + the way we might have scripted our tests to run made it so that pytest could not find specific packages (https://github.com/zooniverse/theia/runs/6310457809?check_suite_focus=true ) Not 100 on this theory. 

 I looked into this a bit and i thnk pytest was doing some sys.path magic and putting theia.something before the system python module path so it was dong a circular module import somewhere. 

Whats super weird is that we should be able to set this on `setup.cfg` but when I did so, it was almost like the tests couldnt read setup.cfg, but it does work if i just set on `python.ini` file. I didn't go too deep into why this was. 🤷‍♀️ At least it works when setting in `python.ini`.

yuenmichelle1 commented:
This worked locally when I was using theia's usgs credentials as testing (see slack https://zooniverse.slack.com/archives/C01E05WQTRD/p1652386614010049?thread_ts=1652372469.662129&cid=C01E05WQTRD for results). 

Since we want researchers in. to test, we could hold off on merging this (in case there is issues when this goes into prod). 
So far from what i've been seeing, prod has been acting different than whats happening locally. I can't tell if it's how i'm testing in prod (i.e. using a project where I'm not set up as owner or something else). I am current looking into these diffs (of events in prod vs events in local dev)
